qiskit/passmanager/flow_controllers.py

```diff
@@ -10,149 +10,330 @@
 # copyright notice, and modified files need to carry a notice indicating
 # that they have been altered from the originals.
 
-"""Pass flow controllers to provide pass iterator conditioned on the property set."""
+"""Built-in pass flow controllers."""
 from __future__ import annotations
-from collections import OrderedDict
-from collections.abc import Sequence
-from typing import Union, List
-import logging
 
-from .base_pass import GenericPass
+import logging
+from collections.abc import Callable, Iterable, Sequence, Generator
+from typing import Type, Any
+
+from qiskit.utils.deprecation import deprecate_func
+from .base_tasks import (
+    BaseController,
+    Task,
+)
+from .propertyset import PropertySet
 from .exceptions import PassManagerError
 
 logger = logging.getLogger(__name__)
 
 
-class FlowController:
-    """Base class for multiple types of working list.
+class FlowControllerLinear(BaseController):
+    """A standard flow controller that runs tasks one after the other."""
 
-    This class is a base class for multiple types of working list. When you iterate on it, it
-    returns the next pass to run.
-    """
-
-    registered_controllers = OrderedDict()
+    def __init__(
+        self,
+        tasks: Task | Iterable[Task] = (),
+        *,
+        options: dict[str, Any] | None = None,
+    ):
+        super().__init__(options)
+
+        if not isinstance(tasks, Sequence):
+            tasks = [tasks]
+        if any(not isinstance(t, Task) for t in tasks):
+            raise PassManagerError("Added tasks are not all valid pass manager task types.")
+        self.tasks: tuple[Task] = tuple(tasks)
+
+    @property
+    @deprecate_func(
+        since="0.26.0",
+        additional_msg="Use .tasks attribute instead.",
+        is_property=True,
+    )
```

Comment:
This recurs everywhere, but the `since` should be "0.45.0", because we use "Qiskit" version numbers not "Terra" numbers now (and the next release will be 0.45.0 for both Qiskit and Terra anyway).

Reply:
Hmm I cannot remember why I chose this number... fixed in [7fba347](https://github.com/Qiskit/qiskit/pull/10127/commits/7fba3479fec357f42a866cd247fe516a0dbd7f01)